File: notification/src/main/java/com/ezbuy/notification/repository/ChannelRepository.java
package com.ezbuy.notification.repository;

import org.springframework.data.r2dbc.repository.Query;
import org.springframework.data.r2dbc.repository.R2dbcRepository;

import com.ezbuy.notification.model.Channel;
import com.ezbuy.notification.repository.query.ChannelQuery;

import reactor.core.publisher.Mono;

public interface ChannelRepository extends R2dbcRepository<Channel, String> {
    Mono<Channel> findById(String id);

    @Query(ChannelQuery.findChannelIdByType)
    Mono<String> findChannelIdByType(String channelType);
}

File: notification-service/src/main/kotlin/com/ezbuy/notificationservice/repository/ChannelRepository.kt
package com.ezbuy.notificationservice.repository

import com.ezbuy.notificationservice.repository.query.ChannelQuery
import com.ezbuy.notificationmodel.model.Channel
import org.springframework.data.r2dbc.repository.R2dbcRepository
import org.springframework.data.r2dbc.repository.Query
import reactor.core.publisher.Mono

interface ChannelRepository : R2dbcRepository<Channel, String> {

    override fun findById(id: String): Mono<Channel>

    @Query(ChannelQuery.FIND_CHANNEL_ID_BY_TYPE)
    fun findChannelIdByType(channelType: String): Mono<String>
}
